ARCHITECTURE.md

```
@@ -0,0 +1,411 @@
+Every operation returns `Result<Success, Failure>`:
+
+```
+Promise<Result<unknown, TrembitaRequestError>>
+                        │
+                        ├─ fetch_failed
+                        │  └─ cause: Error (DNS, connection reset, etc.)
+                        │
+                        ├─ timeout
+                        │  └─ requestedMs: number
+                        │
+                        ├─ circuit_open
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)